packages/front-end/pages/setup/index.tsx

```diff
@@ -0,0 +1,288 @@
+import { useEffect, useState } from "react";
+import {
+  CreateSDKConnectionParams,
+  SDKConnectionInterface,
+  SDKLanguage,
+} from "@back-end/types/sdk-connection";
+import { useForm } from "react-hook-form";
+import { Environment } from "@back-end/types/organization";
+import { getLatestSDKVersion, getSDKCapabilities } from "shared/sdk-versioning";
+import { ProjectInterface } from "@back-end/types/project";
+import PagedModal from "@/components/Modal/PagedModal";
+import { useUser } from "@/services/UserContext";
+import Page from "@/components/Modal/Page";
+import InitiateConnectionPage from "@/components/InitialSetup/InitiateConnectionPage";
+import track from "@/services/track";
+import { useAuth } from "@/services/auth";
+import VerifyConnectionPage from "@/components/InitialSetup/VerifyConnectionPage";
+import SelectDataSourcePage from "@/components/InitialSetup/SelectDataSourcePage";
+import PageHead from "@/components/Layout/PageHead";
+import SetupCompletedPage from "@/components/InitialSetup/SetupCompletedPage";
+import { languageMapping } from "@/components/Features/SDKConnections/SDKLanguageLogo";
+import { useEnvironments } from "@/services/features";
+import usePermissionsUtil from "@/hooks/usePermissionsUtils";
+import { useDefinitions } from "@/services/DefinitionsContext";
+import useSDKConnections from "@/hooks/useSDKConnections";
+
+export type SdkFormValues = {
+  languages: SDKLanguage[];
+  sdkVersion: string;
+  environment: string;
+};
+
+export type ProjectApiResponse = {
+  project: ProjectInterface;
+};
+
+export default function SetupFlow() {
+  const [step, setStep] = useState(0);
+  const [connection, setConnection] = useState<null | string>(null);
+  const [SDKConnectionModalOpen, setSDKConnectionModalOpen] = useState(false);
+  const [setupComplete, setSetupComplete] = useState(false);
+  const [skipped, setSkipped] = useState<Set<number>>(() => new Set());
+
+  const { hasCommercialFeature } = useUser();
+  const { data: sdkConnectionData } = useSDKConnections();
+  const { organization, refreshOrganization } = useUser();
+  const { datasources, setProject, mutateDefinitions } = useDefinitions();
+  const environments = useEnvironments();
+
+  const sdkConnectionForm = useForm<SdkFormValues>({
+    defaultValues: {
+      languages: ["react"],
+      sdkVersion: "",
+      environment: "dev",
+    },
+  });
+
+  // Start off user on the correct step depending on how much has already been set up
+  useEffect(() => {
+    if (!sdkConnectionData?.connections.length) {
+      return;
+    }
+    const firstConnection = sdkConnectionData.connections[0];
```

Comment:
These seems to correctly pick the first SDK connection that was created, but we could also do a reduce here to find the one with the earliest `dateCreated` if needed